packages/experiments/src/components/Button/PrimaryButton.tsx

```diff
@@ -3,7 +3,7 @@ import { Button } from './Button';
 import { ButtonVariantsType } from './ButtonVariants.types';
 
 export const PrimaryButton: ButtonVariantsType = props => {
-  const { text, ...rest } = props;
+  const { text, iconProps, ...rest } = props;
 
-  return <Button primary content={text} {...rest} />;
+  return <Button primary content={text} icon={iconProps} {...rest} />;
```

Comment:
What is the name that appears in the hierarchy for these variants? If it's `Button` then styling and theming targeting `PrimaryButton` by name won't work. If both `Button` and `PrimaryButton` are appearing in hierarchy, then I think styling and theming will be even more wonky.

All of this is just driving need for an `extendComponent` utility that will just have one entry called `PrimaryButton` in the hierarchy that should still take directed styling and theming. This is covered by #8331 (and it seems should be resolved before Fabric 7 release.)

Reply:
Yes, I think only `Button` appears in the hierarchy right now.

Reply:
The other button variants appear as `Unknown`. We could apply `displayName` to them but that won't fix the styling issue since there is no helper like `styled` reading the setting and applying it (the variants are just thin wrappers.)

I'll add a note to #8331. The only other solution is to create new variants using `styled` or `createComponent`.